From e4a357613d92d22adb3657420ac10ce8ccccbcf0 Mon Sep 17 00:00:00 2001
From: Santiago Rossi <santiago_rho_psi@hotmail.com>
Date: Sat, 28 Apr 2012 17:38:26 -0300
Subject: [PATCH 37/40] 4.4.13 Cuenca de Sholazar: Fix para flag en guerra

Temporary fix for related issues with "At war" flag, related to Oracles/Frenzyheart factions
---
 src/server/game/Reputation/ReputationMgr.cpp |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/src/server/game/Reputation/ReputationMgr.cpp b/src/server/game/Reputation/ReputationMgr.cpp
index 94a6590..df0ccac 100755
--- a/src/server/game/Reputation/ReputationMgr.cpp
+++ b/src/server/game/Reputation/ReputationMgr.cpp
@@ -548,7 +548,12 @@ void ReputationMgr::LoadFromDB(PreparedQueryResult result)
 
                 // set atWar for hostile
                 if (GetRank(factionEntry) <= REP_HOSTILE)
-                    SetAtWar(faction, true);
+                {
+						// Hack fix for Frenzyheart (1104) and Oracles (1105), remove peace forced
+						if (factionEntry->ID == 1104 || factionEntry->ID == 1105)
+							faction->Flags &= ~FACTION_FLAG_PEACE_FORCED;
+                        SetAtWar(faction, true);
+				}
 
                 // reset changed flag if values similar to saved in DB
                 if (faction->Flags == dbFactionFlags)
-- 
1.7.8.msysgit.0

